tests/net/ieee802154/l2/src/ieee802154_shell_test.c

```diff
@@ -203,27 +231,26 @@ ZTEST(ieee802154_l2_shell, test_active_scan)
 		0x00, /* Pending Addresses */
 		0x00, 0x00 /* Payload */
 	};
-	struct net_pkt *pkt;
 
-	pkt = net_pkt_rx_alloc_with_buffer(net_iface, sizeof(beacon_pkt), AF_UNSPEC, 0, K_FOREVER);
-	if (!pkt) {
-		NET_ERR("*** No buffer to allocate");
+	net_mgmt_init_event_callback(&scan_cb, scan_result_cb, NET_EVENT_IEEE802154_SCAN_RESULT);
+	net_mgmt_add_event_callback(&scan_cb);
+
+	expected_association_bit = true;
+
+	if (create_and_receive_packet(beacon_pkt, sizeof(beacon_pkt)) < 0) {
 		goto fail;
 	}
 
-	net_pkt_set_ieee802154_lqi(pkt, EXPECTED_COORDINATOR_LQI);
-	net_buf_add_mem(pkt->buffer, beacon_pkt, sizeof(beacon_pkt));
+	test_scan_shell_cmd();
 
-	/* The packet will be placed in the RX queue but not yet handled. */
-	if (net_recv_data(net_iface, pkt) < 0) {
-		NET_ERR("Recv data failed");
-		net_pkt_unref(pkt);
+	/* disable the association permit flag */
+	beacon_pkt[14] = 0x40;
+	expected_association_bit = false;
+
+	if (create_and_receive_packet(beacon_pkt, sizeof(beacon_pkt)) < 0) {
```

Comment:
Well done, this is really sweet and short and easy to read. :-)

Reply:
Thanks :)